src/iris/bin/sync_targets.py

```diff
@@ -263,25 +278,88 @@ def sync_from_oncall(config, engine, purge_old_users=True):
             logger.exception('Failed to update user %s' % username)
             continue
 
-    # sync teams between iris and oncall
-    teams_to_insert = oncall_team_names - iris_team_names
-    teams_to_deactivate = iris_team_names - oncall_team_names
+# sync teams between iris and oncall
 
-    logger.info('Teams to insert (%d)' % len(teams_to_insert))
-    for t in teams_to_insert:
+    # iris_db_oncall_team_ids (team_ids in the oncall_team table)
+    # iris_db_oncall_team_names (names in the oncall_team table)
+    # oncall_team_ids (team_ids from oncall api call)
+    # oncall_team_names (names from oncall api call)
+    # oncall_response_dict_name_key (key value pairs of oncall team names and ids from api call)
+    # oncall_response_dict_id_key same as above but key value inverted
+    # iris_team_names (names from target table)
+    # iris_target_name_id_dict dictionary of target name -> target_id mappings
+    # iris_db_oncall_team_name_id_dict dictionary of oncall name -> oncall team_id mappings
+
+# get all incoming names that match a target check if that target has an entry in oncall table if not make one
+
+    iris_db_oncall_team_names = {name for (name, ) in session.execute('SELECT `name` FROM `oncall_team`')}
+    iris_target_name_id_dict = {name: id for name, id in engine.execute('''SELECT `name`, `id` FROM `target` WHERE `type_id` = %s''', target_types['team'])}
+
+    # up to date target names that don't have an entry in the oncall_team table yet
+    matching_target_names_no_oncall_entry = iris_team_names.intersection(oncall_team_names) - iris_db_oncall_team_names
+
+    for t in matching_target_names_no_oncall_entry:
+        logger.info('Inserting existing team into oncall_team %s' % t)
+        try:
+            engine.execute(oncall_add_sql, (iris_target_name_id_dict[t], t, oncall_response_dict_name_key[t]))
+        except SQLAlchemyError as e:
+            logger.exception('Error inserting oncall_team %s: %s' % (t, e))
+            continue
+
+# rename all mismatching target names
+    iris_db_oncall_team_name_id_dict = {name: id for name, id in engine.execute('''SELECT `name`, `oncall_team_id` FROM `oncall_team`''')}
+    possibly_mismatched_names = iris_team_names - oncall_team_names
+
+    # find teams in the iris database whose names have changed
+    for name in possibly_mismatched_names:
+        if iris_db_oncall_team_name_id_dict.get(name) in oncall_team_ids:
+            # rename team to new name
+            target_id_to_rename = iris_target_name_id_dict[name]
+            # get the oncall response name using the oncall_team team_id
+            new_name = oncall_response_dict_id_key[iris_db_oncall_team_name_id_dict[name]]
+            logger.info('Renaming team %s to %s' % (name, new_name))
```

Comment:
the python logging functions automatically run % against the arguments you give for you, so this could be simplified to:

    logger.info('Renaming team %s to %s', name, new_name)

This applies to all of the other logger calls in this diff